utils.py

```diff
@@ -397,6 +397,95 @@ def image_cell_pressed(button_id, n_cols, image_list, *args):
     zoomed_img = html.Img(src=image_list[img_idx], style=config.IMG_STYLE_ZOOM) if len(image_list) > 0 else EMPTY_IMAGE
     return new_classes + [zoomed_img]
 
+def image_cell_pressed_speed_up(button_id, n_cols, image_list, *args):
+    cell_last_clicked = args[-1]
+    # Grid location of the pressed button
+    # cell_loc = [int(i) for i in re.findall('[0-9]+', button_id)]
+    cell_loc = list(map(int, re.findall('[0-9]+', button_id)))
+
+    # Class name of the pressed button
+    previous_class_clicked = args[N_GRID + cell_loc[1] + cell_loc[0]*COLS_MAX]
+    previous_class_clicked = previous_class_clicked.split(' ')
+    new_classes = list(args[N_GRID:-1])
+    if new_classes:
+        i, j = cell_loc
+        idx = i * COLS_MAX + j
+        if not cell_last_clicked:
+            cell_last_clicked = [0,0]
+
+        if cell_last_clicked != cell_loc:
+            previous_class_idx = cell_last_clicked[1] + cell_last_clicked[0]*COLS_MAX
+            previous_class = args[N_GRID + previous_class_idx]
+            # If it was not previously clicked, this cell just keeps it old class name
+            if 'focus' not in previous_class:
+                new_class = previous_class
+            # In this case, this cell currently holds the "last clicked" status, but it must now yield it to
+            # the newly clicked cell
+            elif 'focus' in previous_class and 'focus' not in previous_class_clicked:
+                new_class = ' '.join(class_toggle_focus(previous_class.split(' ')))
+            new_classes[previous_class_idx] = new_class
+
+        # Toggle the focus according to these rules
+        if 'grouped-off' in previous_class_clicked and 'focus' not in previous_class_clicked:
+            new_class_clicked = class_toggle_grouped(class_toggle_focus(previous_class_clicked))
+        elif 'grouped-off' in previous_class_clicked and 'focus' in previous_class_clicked:
+            new_class_clicked = class_toggle_grouped(previous_class_clicked)
+        elif 'grouped-on' in previous_class_clicked and 'focus' not in previous_class_clicked:
+            new_class_clicked = class_toggle_focus(previous_class_clicked)
+        else:
+            assert 'grouped-on' in previous_class_clicked
+            assert 'focus' in previous_class_clicked
+            new_class_clicked = class_turn_off_keep_delete(class_toggle_grouped(class_toggle_focus(previous_class_clicked)))
+        cell_last_clicked = cell_loc
+        new_class_clicked = ' '.join(new_class_clicked)
+        new_classes[idx] = new_class_clicked
+
+
+    else:
+        cell_last_clicked = None
```

Comment:
Does this `else` clause get used any more?

Reply:
Yeah. I don't think it's needed again. Initially, I thought the initialization would be done in the else.